lib/auth/middleware.go

```diff
@@ -391,23 +403,66 @@ func (a *Middleware) ValidateClientVersion(ctx context.Context, info IdentityInf
 	clientVersion, err := semver.NewVersion(clientVersionString)
 	if err != nil {
 		logger.WithError(err).Warn("Failed to determine client version")
+		a.displayRejectedClientAlert(ctx)
 		if err := info.Conn.Close(); err != nil {
 			logger.WithError(err).Warn("Failed to close client connection")
 		}
+
 		return trace.AccessDenied("client version is unsupported")
 	}
 
 	if clientVersion.LessThan(*a.OldestSupportedVersion) {
 		logger.Info("Terminating connection of client using unsupported version")
+		a.displayRejectedClientAlert(ctx)
+
 		if err := info.Conn.Close(); err != nil {
 			logger.WithError(err).Warn("Failed to close client connection")
 		}
+
 		return trace.AccessDenied("client version is unsupported")
 	}
 
 	return nil
 }
 
+// displayRejectedClientAlert creates an alert to notify admins that
+// unsupported Teleport versions exist in the cluster and are explicitly
+// being denied to prevent causing issues. Alerts are limited to being
+// created once per day to reduce backend writes if there are a large
+// number of unsupported clients constantly being rejected.
+func (a *Middleware) displayRejectedClientAlert(ctx context.Context) {
+	if a.AlertCreator == nil {
+		return
+	}
+
+	now := time.Now()
+	lastAlert := a.lastRejectedAlertTime.Load()
+	if !now.Add(24 * time.Hour).After(time.Unix(lastAlert, 0)) {
```

Comment:
It looks like it is valid to do `time.Unix(0, now.UnixNano())`, so regarding my comment a few lines below, I think this just needs to be `time.Unix(0, lastAlert)`